robottelo/ui/computeresource.py

```diff
@@ -137,10 +135,107 @@ def update(self, name, newname=None, parameter_list=None,
             locators['resource.delete'],
             drop_locator=locators['resource.dropdown'],
         )
+
+    def access_compute_resource(self, res_name):
+        """Access a particular compute resource."""
+        strategy, value = locators['resource.get_by_name']
+        resource = self.search(res_name)
+        if resource is None:
+            raise UINoSuchElementError(
+                'Could not find the resource {0}'.format(res_name))
+        self.click((strategy, value % res_name))
+
+    def list_vms(self, res_name):
+        """Lists vms of a particular compute resource.
+        Note: Currently lists only vms that show up on the first page.
+        """
+        self.access_compute_resource(res_name)
+        self.click(locators['resource.virtual_machines_tab'])
+        vm_elements = self.find_element(locators['resource.vm_list'])
+        return [vm.text for vm in vm_elements]
+
+    def add_image(self, res_name, parameter_list):
+        """Adds an image to a compute resource."""
+        self.access_compute_resource(res_name)
+        self.click(locators['resource.image_add'])
+        self.wait_until_element(locators['resource.image_name'])
+        if parameter_list is None:
+            return
+        for parameter_name, parameter_value, parameter_type in parameter_list:
+            param_locator = '.'.join((
+                'resource.image',
+                (parameter_name.lower()).replace(' ', '_')
+            ))
+            if parameter_type == 'field':
+                self.find_element(
+                    locators[param_locator]
+                ).send_keys(parameter_value)
+            elif parameter_type == 'select':
+                self.select(locators[param_locator], parameter_value)
+        self.click(locators['resource.image_submit'])
+        self.wait_until_element(common_locators['notif.success'])
+
+    def list_images(self, res_name):
+        """Lists images on Compute Resource.
+        Note: Currently lists only images that show up on the first page.
+        """
+        self.access_compute_resource(res_name)
+        self.click(locators['resource.images_tab'])
+        image_elements = self.find_element(locators['resource.image_list'])
+        return [image.text for image in image_elements]
+
+    def vm_action_stop(self, res_name, vm_name, really):
+        """Stops a vm on the compute resource."""
+        self.access_compute_resource(res_name)
+        self.click(locators['resource.virtual_machines_tab'])
+        strategy, value = locators['resource.vm_power_button']
+        button = self.find_element((strategy, value % (res_name, vm_name)))
+        if 'Off' not in button.text:
+            raise UIError(
+                'Could not stop VM {0}. VM is not running'.format(vm_name)
+            )
+        self.click(button)
+        self.handle_alert(really)
+        self.wait_until_element(common_locators['notif.success'])
+
+    def vm_action_start(self, res_name, vm_name):
```

Comment:
Yeah, sure will explore that, but would plan to do that in a different PR, this PR is pending since a long time and want to get this merged ASAP now.

What I would do is , raise a new github issue and assign it to me and @san7ket . This PR once merged would also unblock @san7ket  from raising his PR's. 

Will fix that for sure as part of this, https://github.com/SatelliteQE/robottelo/issues/3754